crates/core/tedge_api/src/health.rs

```diff
@@ -1,12 +1,97 @@
 use std::process;
+use std::sync::Arc;
 
+use crate::mqtt_topics::ServiceTopicId;
 use mqtt_channel::Message;
 use mqtt_channel::PubChannel;
 use mqtt_channel::Topic;
 use mqtt_channel::TopicFilter;
 use serde_json::json;
 use time::OffsetDateTime;
 
+/// Encodes a valid health topic.
+///
+/// Health topics are topics on which messages about health status of services are published. To be
+/// able to send health messages, a health topic needs to be constructed for a given entity.
+// TODO: replace `Arc<str>` with `ServiceTopicId` after we're done with transition to new topics
+#[derive(Debug, Clone, PartialEq, Eq)]
+pub struct ServiceHealthTopic(Arc<str>);
+
+impl ServiceHealthTopic {
+    pub fn new(service: ServiceTopicId) -> Self {
+        // XXX: hardcoded MQTT root
+        ServiceHealthTopic(Arc::from(format!("te/{}/status/health", service.as_str())))
+    }
```

Comment:
In terms of the `new` function, the `service` argument is of type `ServiceTopicId` and that type can be any `+/+/+/+` topic, since in the custom MQTT scheme, for a thing to be a service it has to first register as such, just for `device/device-id/service/service-id` registration to entity store happens automatically.

Reply:
Changed in 4b365c3